src/acceptor.rs

```diff
@@ -0,0 +1,712 @@
+    ///
+    /// Parameters:
+    ///
+    /// acceptor: The rustls_acceptor to read bytes into.
+    /// callback: A function that will perform the actual network I/O.
+    /// userdata: An opaque parameter to be passed directly to `callback`.
+    ///   Note: this is distinct from the `userdata` parameter set with
+    ///   `rustls_connection_set_userdata`.
+    /// out_n: An output parameter. On success, this will be written with
+    ///   the number of bytes `callback` said that it wrote.
+    ///
+    /// Returns:
+    ///
+    /// - 0: Success. You should call rustls_acceptor_accept() next.
+    /// - Any non-zero value: error.
+    ///
+    /// This function passes through return values from callback. Typically
+    /// callback should return an errno value. See rustls_read_callback for
+    /// more details.
+    ///
+    /// Memory and lifetimes:
+    ///
+    /// The rustls_read_callback must be valid to call with the given userdata
+    /// parameter for the duration of this call.
+    #[no_mangle]
+    pub extern "C" fn rustls_acceptor_read_tls(
+        acceptor: *mut rustls_acceptor,
+        callback: rustls_read_callback,
+        userdata: *mut c_void,
+        out_n: *mut size_t,
+    ) -> rustls_io_result {
+        ffi_panic_boundary! {
+            let acceptor: &mut Acceptor = try_mut_from_ptr!(acceptor);
+            if out_n.is_null() {
+                return rustls_io_result(EINVAL);
+            }
+            let callback: ReadCallback = try_callback!(callback);
+
+            let mut reader = CallbackReader { callback, userdata };
+
+            let n_read: usize = match acceptor.read_tls(&mut reader) {
+                Ok(n) => n,
+                Err(e) => return rustls_io_result(e.raw_os_error().unwrap_or(EIO)),
+            };
+            unsafe {
+                *out_n = n_read;
+            }
+
+            rustls_io_result(0)
+        }
+    }
+
+    /// Parse all TLS bytes read so far.  If those bytes make up a ClientHello,
+    /// create a rustls_accepted from them.
+    ///
+    /// Parameters:
+    ///
+    /// acceptor: The rustls_acceptor to access.
+    /// out_accepted: An output parameter. The pointed-to pointer will be set
+    ///   to a new rustls_accepted only when the function returns
+    ///   RUSTLS_RESULT_OK.
+    ///
+    /// Returns:
+    ///
+    /// - RUSTLS_RESULT_OK: a ClientHello has successfully been parsed.
+    ///   A pointer to a newly allocated rustls_accepted has been written to
+    ///   *out_accepted.
+    /// - RUSTLS_RESULT_NOT_READY: a full ClientHello has not yet been read.
+    ///   Read more TLS bytes to continue.
+    /// - Any other rustls_result: the TLS bytes read so far cannot be parsed
+    ///   as a ClientHello, and reading additional bytes won't help.
+    ///
+    /// Memory and lifetimes:
+    ///
+    /// After this method returns RUSTLS_RESULT_OK, the rustls_acceptor is
+    /// still allocated and valid. It needs to be free regardless of success
```

Comment:
Nit: "free" -> "freed". Also, (caution, even more nitpicky) maybe it's clearer to talk about the argument `acceptor` vs the argument of type `rustls_acceptor`?

Reply:
Fixed, thanks.